src/lib/Syntax.hs

```diff
@@ -1857,6 +1892,9 @@ instance SubstE AtomSubstVal Block
 deriving instance Show (Block n)
 deriving via WrapE Block n instance Generic (Block n)
 
+instance BindsOneAtomName (BinderP AtomNameC Type) where
+  boundAtomBinding (_ :> ty) = MiscBound ty
```

Comment:
Instead of using `MiscBound`, would it be better to add another type class that would be required by `binderType`?

Reply:
Yes, that might be better